sys/include/net/gnrc/ipv6/nib.h

```diff
@@ -288,6 +288,19 @@ enum {
     GNRC_IPV6_NIB_ROUTE_INFO_TYPE_NSC,
 };
 
+/**
+ * @brief   Types for GNRC_NETIF_BUS_NIB per-interface message bus
+ */
+typedef enum {
+    /**
+     * @brief   Address registration successful
+     *
+     * The Address registration with the boarder router was successful.
+     * The node now has a global prefix.
+     */
+    GNRC_IPV6_NIB_EVENT_REG_SUCCESS,
```

Comment:
`REG_SUCCESS` is somehow both very unspecific (which reg was success?) and very 6LoWPAN-specific (once you figured out, which reg was success).

Maybe, to also to be able to use this with [address validation in classic DAD](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/nib/_nib-slaac.c#L212-L213), rescope the event to `GNRC_IPV6_NIB_EVENT_ADDR_VALID`? Then this can also be used for the link-local addresses, which [with 6Lo-ND are always assumed to be valid](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/nib/_nib-slaac.c#L75-L79).

Reply:
> Maybe, to also to be able to use this with [address validation in classic DAD](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/nib/_nib-slaac.c#L212-L213), rescope the event to `GNRC_IPV6_NIB_EVENT_ADDR_VALID`?

Sounds good!

> Then this can also be used for the link-local addresses, which [with 6Lo-ND are always assumed to be valid](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/network_layer/ipv6/nib/_nib-slaac.c#L75-L79).

Should those then also generate an event?
I'd then add a pointer to the address as message payload so the receiver can check which address just became valid.

Reply:
> Should those then also generate an event?

IMHO yes!

> I'd then add a pointer to the address as message payload so the receiver can check which address just became valid.

I thought this was already the case... as IPv6 allows multiple addresses per interface, this should have been a given. Question is: which pointer? If we just use the one from the interface array, can we be sure the address does not change between the posting of the event and the handling?

Reply:
> If we just use the one from the interface array, can we be sure the address does not change between the posting of the event and the handling?

Not necessarily - under which conditions could a valid address change though?

If it's the application configuring a different address, we should get away with just documenting that corner case.
If this becomes a use-case, we can add sequence numbers, but I wouldn't want to add code just in case.